_posts/2018-02-05-chaining-http-requests-in-elm.md

```diff
@@ -0,0 +1,340 @@
+    })
+    .then(responseB => {
+        return responseB.json()
+    })
+    .then(user => {
+        if (!user.name) {
+            console.log("unspecified")
+        } else {
+            console.log(user.name)
+        }
+    })
+    .catch(err => {
+        console.error(err)
+    })
+```
+
+It would get a little fancier using `async/await`:
+
+```js
+try {
+    const responseA = await fetch("https://api.github.com/events")
+    const events = await responseA.json()
+    if (events.length == 0) {
+        throw "No events."
+    }
+    const { actor: { login } } = events[0]
+    const responseB = await fetch(`https://api.github.com/users/${login}`)
+    const user = await responseB.json()
+    if (!user.name) {
+        console.log("unspecified")
+    } else {
+        console.log(user.name)
+    }
+} catch (err) {
+    console.error(err)
+}
+```
+
+This is already complicated code to read and understand, and it's tricky to do using Elm as well. Let's see how to achieve the same, understanding exactly what we're doing (we've all blindly copied and pasted code in the past, don't deny).
+
+First, let's write the two requests we need; one for fetching the list of events, the second to obtain a given user's details from her `login`:
+
+```haskell
+import Http
+import Json.Decode as Decode
+
+eventsRequest : Http.Request (List String)
+eventsRequest =
+    Http.get "https://api.github.com/events"
+    (Decode.list (Decode.at [ "actor", "login" ] Decode.string))
+
+nameRequest : String -> Http.Request String
+nameRequest login =
+    Http.get ("https://api.github.com/users/" ++ login)
+        (Decode.at [ "name" ]
+            (Decode.oneOf
+                [ Decode.string
+                , Decode.null "unspecified"
+                ]
+            )
+        )
+```
+
+These two functions return `Http.Request` with the type of data they'll retrieve and decode from the JSON body of their respective responses. `nameRequest` handles the case where Github users don't have entered their full name yet, so the `name` field might be a `null`; as with the JavaScript version, we then default to `"unspecified"`.
+
+That's good but now we need to execute and chain these two requests, the second one depending on the result of the first one, where we retrieve the `actor.login` value of the event object.
+
+Elm is a pure language, meaning you can't have side effects in your functions (a side effect is when functions alter things outside of their scope and use these things: an HTTP request is a *huge* side effect). So your functions must return *something* that represents a given side effect, instead of executing it within the function scope itself. The Elm runtime will be in charge of actually performing the side effect.
+
+In Elm, you're usually gonna use a [Task] to describe future operations. Tasks may succeed or fail (like Promises do in JavaScript), but they need to be turned into [Elm commands] to be actually executed.
```

Comment:
are you sure about the `gonna` here? It feels a bit odd in this context of a technical post.